WordPress/gutenberg
# Site icon: Fix site icon styling to display non-square site icons within a square button

Fixes #37564

## Description

Following on from the site icon feature introduced in #35892 this PR fixes the styling for the site icon in the W button so that it displays non-square site icons within a square button. The fix does the following:

* Switch the W logo button image to use a `div` instead of an `img` tag, and set the image as a background image with background size of `cover`.
* Roll out styling to both the site and post editors

Note: I needed to add the `-1px` on the top margin to get the image to be centred correctly within the border as below:

![image](https://user-images.githubusercontent.com/14988353/147018028-4b15b58b-48d3-4258-8644-eac3b1cabb9b.png)

I _think_ this is because the container is `61px` high instead of `60px` to deal with the grey border, so getting the position exact along with the scale animation is slightly tricky. Let me know if anyone thinks this needs tweaking further!

## How has this been tested?




To quickly add a non-square site icon:

1. Using the Site Logo block, set the block to `Use as site icon` in the sidebar inspector controls.
2. Upload a non-square site logo and update the post/page to save the Icon entity.
3. Load the post editor and test that the non-square icon is displayed as a square in the top left and that when hovered it correctly fills the whole square of the button area
4. Load the site editor and test that the non-square icon is displayed as a square in the top left and that when hovered it correctly fills the whole square of the button area, also when you click on it, the highlighted border should be around the outside of the icon as in the screenshot above
5. Try the above with a few differently shaped logos, including square, tall, and narrow icons

Also, I checked that the styling when no icon is set appears to be unaffected (to test this go to: `/wp-admin/customize.php?autofocus[section]=title_tagline` and clear the existing site icon, then reload the post and site editors)

## Screenshots 

| Before | After | 
| --- | --- |
| ![Kapture 2021-12-22 at 12 00 27](https://user-images.githubusercontent.com/14988353/147017394-21256d74-1336-4a18-85d0-98fc6e645586.gif) | ![Kapture 2021-12-22 at 12 05 58](https://user-images.githubusercontent.com/14988353/147017510-70584b58-2b78-4593-8740-08823bd772d4.gif) |

## Types of changes




Bug fix (non-breaking change which fixes an issue)

## Checklist:
